frontend/theme/index.ts

```diff
@@ -0,0 +1,103 @@
+import { fromEntries } from '../utils'
+
+export const breakpoints = {
+  XS: 600,
+  SM: 786,
+  MD: 1024,
+  LG: 1280,
+} as const
+
+type BreakpointSize = keyof typeof breakpoints
+type Breakpoint = { readonly [key in BreakpointSize]: string }
+
+type Typography =
+  | 'fontFamily'
+  | 'paragraphSM'
+  | 'paragraphMD'
+  | 'paragraphLG'
+  | 'headingOverline'
+  | 'headingSM'
+  | 'headingMD'
+  | 'headingLG'
+  | 'headingXL'
+  | 'displaySM'
+
+export type Theme = {
+  colors: {
+    readonly [key: string]: string
+  }
+  breakpoints: Breakpoint
+  typography: {
+    readonly [key in Typography]: string
+  }
+}
+
+const theme: Theme = {
+  colors: {
+    black: '#393939',
+    white: '#ffffff',
+    gray100: '',
```

Comment:
Since grays are frequently used for borders, backgrounds, text, etc., I think we'd greatly benefit from a range of values. I wasn't exactly sure what to put for these values, cause I wanted to include our current grays and then create the other ones based on those, but I might need your help for that.

What do you think?